paasta_tools/kubernetes_tools.py

```diff
@@ -2226,13 +2233,21 @@ def ensure_namespace(kube_client: KubeClient, namespace: str) -> None:
 
 
 def list_deployments(
-    kube_client: KubeClient, label_selector: str = ""
+    kube_client: KubeClient,
+    label_selector: str = "",
+    service_config: KubernetesDeploymentConfig = None,
```

Comment:
thoughts on unconditionally passing in `service_config`? would help us ensure that we're not missing any callers + get rid of all these if-elses to get the namespace in these helpers

Reply:
that said...this is a little unfortunate: looks like all the operators subclass `LongRunningServiceConfig` directly and not `KubernetesDeploymentConfig` - not sure if we should just go ahead and merge `LongRunningServiceConfig` and `KubernetesDeploymentConfig` (since we only run long-running services on k8s or have the operators (flink, etc) subclass `KubernetesDeploymentConfig`

Reply:
thanks, I removed ``service_config: KubernetesDeploymentConfig = None`` from functions in kubernetes_tools.py and passed ``namespace: str = "paasta"`` instead. I will add ``service_config.get_namespace()`` from the calling code (SKJ) in another PR since it will fall under changing the way skj works